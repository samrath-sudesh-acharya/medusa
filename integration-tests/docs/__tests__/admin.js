const fixtureWriter = require("../utils/write-fixture").default
const { useApi } = require("../../helpers/use-api")
const { useDb } = require("../../helpers/use-db")

const adminSeeder = require("../helpers/admin-seeder")

const customerTest = require("../test-input/admin/customer")
const discountTest = require("../test-input/admin/discount")
const doTest = require("../test-input/admin/draft-order")
const gcTest = require("../test-input/admin/gift-card")
const orderTest = require("../test-input/admin/order")
const productTest = require("../test-input/admin/product")
const rrTest = require("../test-input/admin/return-reason")
const swapTest = require("../test-input/admin/swap")
const collectionTest = require("../test-input/admin/collection")

const toTest = [
  customerTest,
  discountTest,
  doTest,
  gcTest,
  orderTest,
  productTest,
  rrTest,
  swapTest,
  collectionTest,
]

jest.setTimeout(30000)

test.each(toTest)(
  "$operationId",
  async ({ setup, buildEndpoint, snapshotMatch, operationId }) => {
    const api = useApi()
    const db = useDb()
    await adminSeeder(db.connection)

    const manager = db.connection.manager
    const id = await setup(manager, api)

    const response = await api
      .get(buildEndpoint(id), {
        headers: {
          Authorization: "Bearer test_token",
        },
      })
      .catch((err) => {
        console.log(err)
      })

    expect(response.data).toMatchSnapshot(snapshotMatch)

    fixtureWriter.addFixture(`admin.${operationId}`, response.data)
  }
)
