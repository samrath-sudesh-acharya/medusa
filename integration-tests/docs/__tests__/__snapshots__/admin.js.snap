// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetCollection 1`] = `
Object {
  "collection": Object {
    "created_at": Any<String>,
    "deleted_at": null,
    "handle": "spring-2021-collection",
    "id": StringMatching /\\^pcol_\\*/,
    "metadata": null,
    "title": Any<String>,
    "updated_at": Any<String>,
  },
}
`;

exports[`GetCustomersCustomer 1`] = `
Object {
  "customer": Object {
    "billing_address": null,
    "billing_address_id": null,
    "created_at": Any<String>,
    "deleted_at": null,
    "email": "test1@email.com",
    "first_name": null,
    "has_account": false,
    "id": StringMatching /\\^cus_\\*/,
    "last_name": null,
    "metadata": null,
    "orders": Array [],
    "phone": null,
    "shipping_addresses": Array [],
    "updated_at": Any<String>,
  },
}
`;

exports[`GetDiscountsDiscount 1`] = `
Object {
  "discount": Object {
    "code": "ABCD-1234-FIVE-QWER",
    "created_at": Any<String>,
    "deleted_at": null,
    "ends_at": null,
    "id": StringMatching /\\^disc_\\*/,
    "is_disabled": false,
    "is_dynamic": false,
    "metadata": null,
    "parent_discount": null,
    "parent_discount_id": null,
    "regions": Array [
      Object {
        "created_at": Any<String>,
        "currency_code": "dkk",
        "deleted_at": null,
        "fulfillment_providers": Array [
          Object {
            "id": "test-ful",
            "is_installed": true,
          },
        ],
        "id": StringMatching /\\^reg_\\*/,
        "metadata": null,
        "name": "Scandinavia",
        "payment_providers": Array [
          Object {
            "id": "test-pay",
            "is_installed": true,
          },
        ],
        "tax_code": null,
        "tax_rate": "0",
        "updated_at": Any<String>,
      },
    ],
    "rule": null,
    "rule_id": null,
    "starts_at": Any<String>,
    "updated_at": Any<String>,
    "usage_count": 0,
    "usage_limit": 5,
  },
}
`;

exports[`GetDraftOrdersDraftOrder 1`] = `
Object {
  "draft_order": Object {
    "canceled_at": null,
    "cart": Object {
      "billing_address": null,
      "billing_address_id": null,
      "completed_at": null,
      "context": null,
      "created_at": Any<String>,
      "customer_id": StringMatching /\\^cus_\\*/,
      "deleted_at": null,
      "discount_total": 0,
      "discounts": Array [],
      "email": "jane@medusa.test",
      "gift_card_total": 0,
      "gift_cards": Array [],
      "id": StringMatching /\\^cart_\\*/,
      "idempotency_key": null,
      "items": Array [
        Object {
          "allow_discounts": true,
          "cart_id": StringMatching /\\^cart_\\*/,
          "claim_order_id": null,
          "created_at": Any<String>,
          "description": "Soft and sweet",
          "fulfilled_quantity": 1,
          "has_shipping": null,
          "id": StringMatching /\\^item_\\*/,
          "is_giftcard": false,
          "metadata": null,
          "order_id": null,
          "quantity": 1,
          "returned_quantity": null,
          "shipped_quantity": null,
          "should_merge": true,
          "swap_id": null,
          "thumbnail": "https://test.js/1234",
          "title": "Basic Tee - One Size",
          "unit_price": 8000,
          "updated_at": Any<String>,
          "variant": Object {
            "allow_backorder": false,
            "barcode": null,
            "created_at": Any<String>,
            "deleted_at": null,
            "ean": null,
            "height": null,
            "hs_code": null,
            "id": StringMatching /\\^variant_\\*/,
            "inventory_quantity": 1,
            "length": null,
            "manage_inventory": true,
            "material": null,
            "metadata": null,
            "mid_code": null,
            "origin_country": null,
            "product_id": StringMatching /\\^prod_\\*/,
            "sku": null,
            "title": "One Size",
            "upc": null,
            "updated_at": Any<String>,
            "weight": null,
            "width": null,
          },
          "variant_id": StringMatching /\\^variant_\\*/,
        },
      ],
      "metadata": null,
      "payment": null,
      "payment_authorized_at": null,
      "payment_id": null,
      "payment_sessions": Array [],
      "region": Object {
        "created_at": Any<String>,
        "currency_code": "usd",
        "deleted_at": null,
        "fulfillment_providers": Array [],
        "id": StringMatching /\\^reg_\\*/,
        "metadata": null,
        "name": "United States",
        "payment_providers": Array [
          Object {
            "id": "test-pay",
            "is_installed": true,
          },
        ],
        "tax_code": null,
        "tax_rate": "0",
        "updated_at": Any<String>,
      },
      "region_id": StringMatching /\\^reg_\\*/,
      "shipping_address": Object {
        "address_1": null,
        "address_2": null,
        "city": null,
        "company": null,
        "country_code": "dk",
        "created_at": Any<String>,
        "customer_id": null,
        "deleted_at": null,
        "first_name": "Jane",
        "id": StringMatching /\\^addr_\\*/,
        "last_name": "Medusan",
        "metadata": null,
        "phone": null,
        "postal_code": null,
        "province": null,
        "updated_at": Any<String>,
      },
      "shipping_address_id": StringMatching /\\^addr_\\*/,
      "shipping_methods": Array [],
      "shipping_total": 0,
      "subtotal": 8000,
      "tax_total": 0,
      "total": 8000,
      "type": "draft_order",
      "updated_at": Any<String>,
    },
    "cart_id": StringMatching /\\^cart_\\*/,
    "created_at": Any<String>,
    "display_id": 1,
    "id": StringMatching /\\^dorder_\\*/,
    "metadata": null,
    "no_notification_order": null,
    "order": null,
    "order_id": null,
    "status": "open",
    "updated_at": Any<String>,
  },
}
`;

exports[`GetGiftCardsGiftCard 1`] = `
Object {
  "gift_card": Object {
    "balance": 1000,
    "code": Any<String>,
    "created_at": Any<String>,
    "deleted_at": null,
    "ends_at": null,
    "id": StringMatching /\\^gift_\\*/,
    "is_disabled": false,
    "metadata": null,
    "order": null,
    "region": Object {
      "created_at": Any<String>,
      "currency_code": "usd",
      "deleted_at": null,
      "fulfillment_providers": Array [],
      "id": StringMatching /\\^reg_\\*/,
      "metadata": null,
      "name": "United States",
      "payment_providers": Array [],
      "tax_code": null,
      "tax_rate": "0",
      "updated_at": Any<String>,
    },
    "region_id": StringMatching /\\^reg_\\*/,
    "updated_at": Any<String>,
    "value": 1000,
  },
}
`;

exports[`GetOrdersOrder 1`] = `
Object {
  "order": Object {
    "billing_address": null,
    "canceled_at": null,
    "cart_id": null,
    "claims": Array [],
    "created_at": Any<String>,
    "currency_code": "usd",
    "customer": Object {
      "billing_address_id": null,
      "created_at": Any<String>,
      "deleted_at": null,
      "email": "jane@medusa.test",
      "first_name": null,
      "has_account": false,
      "id": StringMatching /\\^cus_\\*/,
      "last_name": null,
      "metadata": null,
      "password_hash": null,
      "phone": null,
      "updated_at": Any<String>,
    },
    "customer_id": StringMatching /\\^cus_\\*/,
    "discount_total": 800,
    "discounts": Array [
      Object {
        "code": "TEST134",
        "created_at": Any<String>,
        "deleted_at": null,
        "ends_at": null,
        "id": StringMatching /\\^disc_\\*/,
        "is_disabled": false,
        "is_dynamic": false,
        "metadata": null,
        "parent_discount_id": null,
        "rule": Object {
          "allocation": "total",
          "created_at": Any<String>,
          "deleted_at": null,
          "description": "Test Discount",
          "id": StringMatching /\\^dru_\\*/,
          "metadata": null,
          "type": "percentage",
          "updated_at": Any<String>,
          "value": 10,
        },
        "rule_id": StringMatching /\\^dru_\\*/,
        "starts_at": Any<String>,
        "updated_at": Any<String>,
        "usage_count": 0,
        "usage_limit": null,
      },
    ],
    "display_id": 1,
    "draft_order_id": null,
    "email": "jane@medusa.test",
    "fulfillment_status": "fulfilled",
    "fulfillments": Array [],
    "gift_card_total": 0,
    "gift_card_transactions": Array [],
    "gift_cards": Array [],
    "id": StringMatching /\\^order_\\*/,
    "items": Array [
      Object {
        "allow_discounts": true,
        "cart_id": null,
        "claim_order_id": null,
        "created_at": Any<String>,
        "description": "Line Item Desc",
        "fulfilled_quantity": 1,
        "has_shipping": null,
        "id": StringMatching /\\^item_\\*/,
        "is_giftcard": false,
        "metadata": null,
        "order_id": StringMatching /\\^order_\\*/,
        "quantity": 1,
        "refundable": 7200,
        "returned_quantity": 0,
        "shipped_quantity": null,
        "should_merge": true,
        "swap_id": null,
        "thumbnail": "https://test.js/1234",
        "title": "Line Item",
        "unit_price": 8000,
        "updated_at": Any<String>,
        "variant": Object {
          "allow_backorder": false,
          "barcode": null,
          "created_at": Any<String>,
          "deleted_at": null,
          "ean": null,
          "height": null,
          "hs_code": null,
          "id": StringMatching /\\^variant_\\*/,
          "inventory_quantity": 1,
          "length": null,
          "manage_inventory": true,
          "material": null,
          "metadata": null,
          "mid_code": null,
          "origin_country": null,
          "product_id": StringMatching /\\^prod_\\*/,
          "sku": null,
          "title": "One Size",
          "upc": null,
          "updated_at": Any<String>,
          "weight": null,
          "width": null,
        },
        "variant_id": StringMatching /\\^variant_\\*/,
      },
    ],
    "metadata": null,
    "no_notification": null,
    "paid_total": 10000,
    "payment_status": "captured",
    "payments": Array [
      Object {
        "amount": 10000,
        "amount_refunded": 0,
        "canceled_at": null,
        "captured_at": null,
        "cart_id": null,
        "created_at": Any<String>,
        "currency_code": "usd",
        "data": Object {},
        "id": StringMatching /\\^pay_\\*/,
        "idempotency_key": null,
        "metadata": null,
        "order_id": StringMatching /\\^order_\\*/,
        "provider_id": "test-pay",
        "swap_id": null,
        "updated_at": Any<String>,
      },
    ],
    "refundable_amount": 10000,
    "refunded_total": 0,
    "refunds": Array [],
    "region": Object {
      "created_at": Any<String>,
      "currency_code": "usd",
      "deleted_at": null,
      "fulfillment_providers": Array [
        Object {
          "id": "test-ful",
          "is_installed": true,
        },
      ],
      "id": StringMatching /\\^reg_\\*/,
      "metadata": null,
      "name": "United States",
      "payment_providers": Array [
        Object {
          "id": "test-pay",
          "is_installed": true,
        },
      ],
      "tax_code": null,
      "tax_rate": "0",
      "updated_at": Any<String>,
    },
    "region_id": StringMatching /\\^reg_\\*/,
    "returns": Array [],
    "shipping_address": Object {
      "address_1": null,
      "address_2": null,
      "city": null,
      "company": null,
      "country_code": "us",
      "created_at": Any<String>,
      "customer_id": null,
      "deleted_at": null,
      "first_name": "Jane",
      "id": StringMatching /\\^addr_\\*/,
      "last_name": "Medusan",
      "metadata": null,
      "phone": null,
      "postal_code": null,
      "province": null,
      "updated_at": Any<String>,
    },
    "shipping_methods": Array [
      Object {
        "cart_id": null,
        "claim_order_id": null,
        "data": Object {},
        "id": StringMatching /\\^sm_\\*/,
        "order_id": StringMatching /\\^order_\\*/,
        "price": 1000,
        "return_id": null,
        "shipping_option": Object {
          "admin_only": false,
          "amount": 1000,
          "created_at": Any<String>,
          "data": Object {},
          "deleted_at": null,
          "id": StringMatching /\\^so_\\*/,
          "is_return": false,
          "metadata": null,
          "name": "Standard Shipping",
          "price_type": "flat_rate",
          "profile_id": StringMatching /\\^sp_\\*/,
          "provider_id": "test-ful",
          "region_id": StringMatching /\\^reg_\\*/,
          "updated_at": Any<String>,
        },
        "shipping_option_id": StringMatching /\\^so_\\*/,
        "swap_id": null,
      },
    ],
    "shipping_total": 1000,
    "status": "pending",
    "subtotal": 8000,
    "swaps": Array [],
    "tax_rate": 0,
    "tax_total": 0,
    "total": 8200,
    "updated_at": Any<String>,
  },
}
`;

exports[`GetProductsProduct 1`] = `
Object {
  "product": Object {
    "collection": null,
    "collection_id": null,
    "created_at": Any<String>,
    "deleted_at": null,
    "description": null,
    "discountable": true,
    "handle": "basic-tee",
    "height": null,
    "hs_code": null,
    "id": StringMatching /\\^prod_\\*/,
    "images": Array [],
    "is_giftcard": false,
    "length": null,
    "material": null,
    "metadata": null,
    "mid_code": null,
    "options": Array [
      Object {
        "created_at": Any<String>,
        "deleted_at": null,
        "id": StringMatching /\\^opt_\\*/,
        "metadata": null,
        "product_id": StringMatching /\\^prod_\\*/,
        "title": "Size",
        "updated_at": Any<String>,
      },
    ],
    "origin_country": null,
    "profile_id": StringMatching /\\^sp_\\*/,
    "status": "draft",
    "subtitle": null,
    "tags": Array [],
    "thumbnail": null,
    "title": "Basic Tee",
    "type": null,
    "type_id": null,
    "updated_at": Any<String>,
    "variants": Array [
      Object {
        "allow_backorder": false,
        "barcode": null,
        "created_at": Any<String>,
        "deleted_at": null,
        "ean": null,
        "height": null,
        "hs_code": null,
        "id": StringMatching /\\^variant_\\*/,
        "inventory_quantity": 1,
        "length": null,
        "manage_inventory": true,
        "material": null,
        "metadata": null,
        "mid_code": null,
        "options": Array [
          Object {
            "created_at": Any<String>,
            "deleted_at": null,
            "id": StringMatching /\\^optval_\\*/,
            "metadata": null,
            "option_id": StringMatching /\\^opt_\\*/,
            "updated_at": Any<String>,
            "value": "Size",
            "variant_id": StringMatching /\\^variant_\\*/,
          },
        ],
        "origin_country": null,
        "prices": Array [
          Object {
            "amount": 8000,
            "created_at": Any<String>,
            "currency_code": "usd",
            "deleted_at": null,
            "id": StringMatching /\\^ma_\\*/,
            "region_id": null,
            "sale_amount": null,
            "updated_at": Any<String>,
            "variant_id": StringMatching /\\^variant_\\*/,
          },
        ],
        "product_id": StringMatching /\\^prod_\\*/,
        "sku": null,
        "title": "One Size",
        "upc": null,
        "updated_at": Any<String>,
        "weight": null,
        "width": null,
      },
    ],
    "weight": null,
    "width": null,
  },
}
`;

exports[`GetReturnReasonsReason 1`] = `
Object {
  "return_reason": Object {
    "created_at": Any<String>,
    "deleted_at": null,
    "description": "Use this if the size was too big",
    "id": StringMatching /\\^rr_\\*/,
    "label": "Too Big",
    "updated_at": Any<String>,
    "value": "too_big",
  },
}
`;

exports[`GetSwapsSwap 1`] = `
Object {
  "swap": Object {
    "additional_items": Array [
      Object {
        "allow_discounts": true,
        "cart_id": StringMatching /\\^cart_\\*/,
        "claim_order_id": null,
        "created_at": Any<String>,
        "description": "Large",
        "fulfilled_quantity": null,
        "has_shipping": null,
        "id": StringMatching /\\^item_\\*/,
        "is_giftcard": false,
        "metadata": Object {},
        "order_id": null,
        "quantity": 1,
        "returned_quantity": null,
        "shipped_quantity": null,
        "should_merge": true,
        "swap_id": StringMatching /\\^swap_\\*/,
        "thumbnail": null,
        "title": "Basic Tee",
        "unit_price": 8000,
        "updated_at": Any<String>,
        "variant": Object {
          "allow_backorder": false,
          "barcode": null,
          "created_at": Any<String>,
          "deleted_at": null,
          "ean": null,
          "height": null,
          "hs_code": null,
          "id": StringMatching /\\^variant_\\*/,
          "inventory_quantity": 1,
          "length": null,
          "manage_inventory": true,
          "material": null,
          "metadata": null,
          "mid_code": null,
          "origin_country": null,
          "product_id": StringMatching /\\^prod_\\*/,
          "sku": null,
          "title": "Large",
          "upc": null,
          "updated_at": Any<String>,
          "weight": null,
          "width": null,
        },
        "variant_id": StringMatching /\\^variant_\\*/,
      },
    ],
    "cart": Object {
      "billing_address_id": null,
      "completed_at": null,
      "context": null,
      "created_at": Any<String>,
      "customer_id": StringMatching /\\^cus_\\*/,
      "deleted_at": null,
      "discount_total": 0,
      "discounts": Array [
        Object {
          "code": "TEST134",
          "created_at": Any<String>,
          "deleted_at": null,
          "ends_at": null,
          "id": StringMatching /\\^disc_\\*/,
          "is_disabled": false,
          "is_dynamic": false,
          "metadata": null,
          "parent_discount_id": null,
          "rule": Object {
            "allocation": "total",
            "created_at": Any<String>,
            "deleted_at": null,
            "description": "Test Discount",
            "id": StringMatching /\\^dru_\\*/,
            "metadata": null,
            "type": "percentage",
            "updated_at": Any<String>,
            "value": 10,
          },
          "rule_id": StringMatching /\\^dru_\\*/,
          "starts_at": Any<String>,
          "updated_at": Any<String>,
          "usage_count": 0,
          "usage_limit": null,
        },
      ],
      "email": "jane@medusa.com",
      "gift_card_total": 0,
      "gift_cards": Array [],
      "id": StringMatching /\\^cart_\\*/,
      "idempotency_key": null,
      "items": Array [
        Object {
          "allow_discounts": true,
          "cart_id": StringMatching /\\^cart_\\*/,
          "claim_order_id": null,
          "created_at": Any<String>,
          "description": "Large",
          "fulfilled_quantity": null,
          "has_shipping": null,
          "id": StringMatching /\\^item_\\*/,
          "is_giftcard": false,
          "metadata": Object {},
          "order_id": null,
          "quantity": 1,
          "returned_quantity": null,
          "shipped_quantity": null,
          "should_merge": true,
          "swap_id": StringMatching /\\^swap_\\*/,
          "thumbnail": null,
          "title": "Basic Tee",
          "unit_price": 8000,
          "updated_at": Any<String>,
          "variant": Object {
            "allow_backorder": false,
            "barcode": null,
            "created_at": Any<String>,
            "deleted_at": null,
            "ean": null,
            "height": null,
            "hs_code": null,
            "id": StringMatching /\\^variant_\\*/,
            "inventory_quantity": 1,
            "length": null,
            "manage_inventory": true,
            "material": null,
            "metadata": null,
            "mid_code": null,
            "origin_country": null,
            "product_id": StringMatching /\\^prod_\\*/,
            "sku": null,
            "title": "Large",
            "upc": null,
            "updated_at": Any<String>,
            "weight": null,
            "width": null,
          },
          "variant_id": StringMatching /\\^variant_\\*/,
        },
        Object {
          "allow_discounts": true,
          "cart_id": StringMatching /\\^cart_\\*/,
          "claim_order_id": null,
          "created_at": Any<String>,
          "description": null,
          "fulfilled_quantity": null,
          "has_shipping": null,
          "id": StringMatching /\\^item_\\*/,
          "is_giftcard": false,
          "metadata": Object {
            "is_return_line": true,
          },
          "order_id": null,
          "quantity": 1,
          "returned_quantity": null,
          "shipped_quantity": null,
          "should_merge": true,
          "swap_id": null,
          "thumbnail": "https://test.js/1234",
          "title": "Line Item",
          "unit_price": -8000,
          "updated_at": Any<String>,
          "variant": Object {
            "allow_backorder": false,
            "barcode": null,
            "created_at": Any<String>,
            "deleted_at": null,
            "ean": null,
            "height": null,
            "hs_code": null,
            "id": StringMatching /\\^variant_\\*/,
            "inventory_quantity": 1,
            "length": null,
            "manage_inventory": true,
            "material": null,
            "metadata": null,
            "mid_code": null,
            "origin_country": null,
            "product_id": StringMatching /\\^prod_\\*/,
            "sku": null,
            "title": "Small",
            "upc": null,
            "updated_at": Any<String>,
            "weight": null,
            "width": null,
          },
          "variant_id": StringMatching /\\^variant_\\*/,
        },
      ],
      "metadata": Object {
        "parent_order_id": StringMatching /\\^order_\\*/,
        "swap_id": StringMatching /\\^swap_\\*/,
      },
      "payment": null,
      "payment_authorized_at": null,
      "payment_id": null,
      "region": Object {
        "created_at": Any<String>,
        "currency_code": "usd",
        "deleted_at": null,
        "fulfillment_providers": Array [
          Object {
            "id": "test-ful",
            "is_installed": true,
          },
        ],
        "id": StringMatching /\\^reg_\\*/,
        "metadata": null,
        "name": "United States",
        "payment_providers": Array [
          Object {
            "id": "test-pay",
            "is_installed": true,
          },
        ],
        "tax_code": null,
        "tax_rate": "0",
        "updated_at": Any<String>,
      },
      "region_id": StringMatching /\\^reg_\\*/,
      "shipping_address_id": StringMatching /\\^addr_\\*/,
      "shipping_methods": Array [],
      "shipping_total": 0,
      "subtotal": 0,
      "tax_total": 0,
      "total": 0,
      "type": "swap",
      "updated_at": Any<String>,
    },
    "cart_id": StringMatching /\\^cart_\\*/,
    "created_at": Any<String>,
    "difference_due": null,
    "fulfillment_status": "not_fulfilled",
    "fulfillments": Array [],
    "id": StringMatching /\\^swap_\\*/,
    "metadata": null,
    "order": Object {
      "billing_address_id": null,
      "canceled_at": null,
      "cart_id": null,
      "created_at": Any<String>,
      "currency_code": "usd",
      "customer_id": StringMatching /\\^cus_\\*/,
      "display_id": 1,
      "draft_order_id": null,
      "email": "jane@medusa.com",
      "fulfillment_status": "fulfilled",
      "id": StringMatching /\\^order_\\*/,
      "idempotency_key": null,
      "metadata": null,
      "no_notification": null,
      "payment_status": "captured",
      "region_id": StringMatching /\\^reg_\\*/,
      "shipping_address_id": StringMatching /\\^addr_\\*/,
      "status": "pending",
      "tax_rate": 0,
      "updated_at": Any<String>,
    },
    "order_id": StringMatching /\\^order_\\*/,
    "payment": null,
    "payment_status": "not_paid",
    "return_order": Object {
      "claim_order_id": null,
      "created_at": Any<String>,
      "id": StringMatching /\\^ret_\\*/,
      "idempotency_key": null,
      "items": Array [
        Object {
          "is_requested": true,
          "item_id": StringMatching /\\^item_\\*/,
          "metadata": null,
          "note": null,
          "quantity": 1,
          "reason_id": null,
          "received_quantity": null,
          "requested_quantity": 1,
          "return_id": StringMatching /\\^ret_\\*/,
        },
      ],
      "metadata": null,
      "no_notification": null,
      "order_id": null,
      "received_at": null,
      "refund_amount": 7200,
      "shipping_data": null,
      "status": "requested",
      "swap_id": StringMatching /\\^swap_\\*/,
      "updated_at": Any<String>,
    },
    "shipping_address": null,
    "shipping_methods": Array [],
    "updated_at": Any<String>,
  },
}
`;
